src/rewriters/Rewriter.h

```diff
@@ -46,30 +48,29 @@ class Rewriter {
                 continue;
             }
             // If we are here, we have already processed all children
-            assert(done[Idx(term.getId())] == 0);
             vec<PTRef> newArgs(childrenCount);
             bool needsChange = false;
-            for (unsigned i = 0; i < childrenCount; ++i) {
-                auto it = substitutions.find(term[i]);
-                bool childChanged = it != substitutions.end();
+            for (auto i = 0; i < newArgs.size(); ++i) {
+                PTRef it;
```

Comment:
Maybe this could have a better name now, since it is no longer an iterator.

Reply:
Fixed.